core/ROB.cpp

```diff
@@ -171,6 +171,14 @@ namespace olympia
                     break;
                 }
 
+                // Is this a misprdicted branch requiring a refetch?
+                if(ex_inst.isMispredicted()) {
+                    FlushManager::FlushingCriteria criteria(FlushManager::FlushCause::MISPREDICTION,
+                                                            ex_inst_ptr);
+                    out_retire_flush_.send(criteria);
+                    break;
+                }
+
```

Comment:
This is going to hit a similar problem right?

You're relying on the flushmanager to flush the ROB before retireInstructions_ is called again?

Reply:
No, the loop breaks right after detecting the mispredicted load.  Another retiring event is scheduled for next cycle which will be called _after_ the flush occurs.

Reply:
It suffers from the same frailties as the previous way right?

You're just moving the race condition from the executepipe to the ROB?

Reply:
You are correct there still is a race condition.  I've added the required delay by breaking the loop and expecting a flush the next cycle.  I'll add code to ensure retirement "stops" until the flush is received.